src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -348,7 +352,7 @@ public function setOptions(array $options)
         ));
 
         foreach ($options as $key => $value) {
-            if (isset($validOptions[$key])) {
+            if (isset($validOptions[$key]) && false === headers_sent()) {
```

Comment:
this should not be checked for each option, but only once before the loop.
And it should use `!header_sent()`